components/infobox/wikis/arenafps/infobox_league_custom.lua

```diff
@@ -0,0 +1,169 @@
+	['macguffin'] = 'MacGuffin',
+	['slipgate'] = 'Slipgate',
+	['clan arena'] = 'Clan Arena',
+}
+
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_league = league
+	_args = _league.args
+
+	_args.player_number = _args.participants_number
+	_args.game = Game.name{game = _args.game}
+	_args.mode = CustomLeague:_modeLookup(_args.mode)
+
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+	league.addToLpdb = CustomLeague.addToLpdb
+	league.getWikiCategories = CustomLeague.getWikiCategories
+
+	return league:createInfobox()
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets, Cell{
+		name = 'Game',
+		content = {_args.game}
+	})
+	table.insert(widgets, Cell{
+		name = 'Mode',
+		content = {_args.mode}
+	})
+	table.insert(widgets, Cell{
+		name = 'Number of Players',
+		content = {_args.player_number}
+	})
+	table.insert(widgets, Cell{
+		name = 'Number of Teams',
+		content = {_args.team_number}
+	})
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'customcontent' then
+		local maps = _league:getAllArgsForBase(_args, 'map')
+		if #maps > 0 then
+			local game = _args.game and ('/' .. _args.game) or ''
+
+			maps = Array.map(maps, function(map)
+				return tostring(CustomLeague:_createNoWrappingSpan(
+					PageLink.makeInternalLink({}, map, map .. game)
+				))
+			end)
+
+			table.insert(widgets, Title{name = 'Maps'})
+			table.insert(widgets, Center{content = {table.concat(maps, '&nbsp;• ')}})
+		end
+	end
+	return widgets
+end
+
+function CustomLeague:addToLpdb(lpdbData, args)
+	lpdbData.maps = table.concat(_league:getAllArgsForBase(args, 'map'), ';')
+	lpdbData.participantsnumber = args.player_number or args.team_number
+
+	return lpdbData
+end
+
+function CustomLeague:defineCustomPageVariables()
+	--Legacy vars
+	Variables.varDefine('tournament_ticker_name', _args.tickername or _args.name)
+	Variables.varDefine('tournament_tier', _args.liquipediatier)
+	Variables.varDefine('tournament_prizepool', _args.prizepoolusd)
+	Variables.varDefine('tournament_mode', _args.mode)
+
+	--Legacy date vars
+	local sdate = Variables.varDefault('tournament_startdate', '')
+	local edate = Variables.varDefault('tournament_enddate', '')
+	Variables.varDefine('tournament_sdate', sdate)
+	Variables.varDefine('tournament_edate', edate)
+	Variables.varDefine('tournament_date', edate)
+	Variables.varDefine('date', edate)
+	Variables.varDefine('sdate', sdate)
+	Variables.varDefine('edate', edate)
+	Variables.varDefine('mode', _args.mode)
+end
+
+function CustomLeague:getWikiCategories(args)
+	local categories = {}
+
+	if Game.name{game = _args.game} then
+		table.insert(categories, Game.name{game = _args.game} .. ' Competitions')
+	else
+		table.insert(categories, 'Tournaments without game version')
+	end
+
+	if _args.mode then
```

Comment:
maybe use `CustomLeague:_modeLookup(mode)` to also catch invalid modes?

Reply:
arg.mode is normalized through modeLookup already